vochain/transaction/account_tx.go

```diff
@@ -239,3 +240,133 @@ func (t *TransactionHandler) SetAccountInfoTxCheck(vtx *vochaintx.Tx) error {
+func (t *TransactionHandler) SetSikTxCheck(vtx *vochaintx.Tx) (common.Address, vstate.SIK, error) {
+	if vtx == nil || vtx.Signature == nil || vtx.SignedBody == nil || vtx.Tx == nil {
+		return common.Address{}, nil, ErrNilTx
+	}
+	tx := vtx.Tx.GetSetSik()
+	if tx == nil {
+		return common.Address{}, nil, fmt.Errorf("invalid transaction")
+	}
+	pubKey, err := ethereum.PubKeyFromSignature(vtx.SignedBody, vtx.Signature)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract public key from vtx.Signature: %w", err)
+	}
+	txAddress, err := ethereum.AddrFromPublicKey(pubKey)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract address from public key: %w", err)
+	}
+
+	if _, err := t.state.GetAccount(txAddress, false); err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot get tx account: %w", err)
+	}
+
+	newSik := vtx.Tx.GetSetSik().GetSik()
+	if newSik == nil {
+		return common.Address{}, nil, fmt.Errorf("no sik value provided")
+	}
+	// check if the address already has invalidated sik to ensure that it is
+	// not updated after reach the correct height to avoid double voting
+	if currentSik, err := t.state.SIKFromAddress(txAddress); err == nil {
+		maxEndBlock, err := t.state.ProcessBlockRegistry.MaxEndBlock(t.state.CurrentHeight(), false)
+		if err != nil {
+			return common.Address{}, nil, err
+		}
+		if height := currentSik.DecodeInvalidatedHeight(); height >= maxEndBlock {
+			return txAddress, nil, fmt.Errorf("the sik could not be changed yet")
+		}
+	}
+	return txAddress, newSik, nil
+}
+
+// RegisterSikTxCheck checks if the provided RegisterSikTx is valid ensuring
+// that the proof included on it is valid for the address of the transaction
+// signer.
+func (t *TransactionHandler) RegisterSikTxCheck(vtx *vochaintx.Tx) (common.Address, vstate.SIK, error) {
+	if vtx == nil || vtx.Signature == nil || vtx.SignedBody == nil || vtx.Tx == nil {
+		return common.Address{}, nil, ErrNilTx
+	}
+	tx := vtx.Tx.GetRegisterSik()
+	if tx == nil {
+		return common.Address{}, nil, fmt.Errorf("invalid transaction")
+	}
+	pubKey, err := ethereum.PubKeyFromSignature(vtx.SignedBody, vtx.Signature)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract public key from vtx.Signature: %w", err)
+	}
+	txAddress, err := ethereum.AddrFromPublicKey(pubKey)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot extract address from public key: %w", err)
+	}
+	if _, err := t.state.GetAccount(txAddress, false); err != nil {
+		return common.Address{}, nil, fmt.Errorf("cannot get the account for the tx address: %w", err)
+	}
+	newSik := tx.GetSik()
+	if newSik == nil {
+		return common.Address{}, nil, fmt.Errorf("no sik value provided")
+	}
+	// check if the address already has invalidated sik to ensure that it is
+	// not updated after reach the correct height to avoid double voting
+	if currentSik, err := t.state.SIKFromAddress(txAddress); err == nil {
```

Comment:
I'd not allow to change the SIK on this function, to simplify and reduce potential attack vectors. I'd just allow register if the SIK does not exist at all (not even invalidated).